src/engine/QueryPlanner.cpp

```diff
@@ -1693,6 +1713,47 @@ std::vector<QueryPlanner::SubtreePlan> QueryPlanner::createJoinCandidates(
   return candidates;
 }
 
+// _____________________________________________________________________________
+auto QueryPlanner::createSpatialJoin(
+    const SubtreePlan& a, const SubtreePlan& b,
+    const std::vector<std::array<ColumnIndex, 2>>& jcs)
+    -> std::optional<SubtreePlan> {
+  auto aIsSpatialJoin =
+      std::dynamic_pointer_cast<const SpatialJoin>(a._qet->getRootOperation());
+  auto bIsSpatialJoin =
+      std::dynamic_pointer_cast<const SpatialJoin>(b._qet->getRootOperation());
+
+  bool aIs = static_cast<bool>(aIsSpatialJoin);
+  bool bIs = static_cast<bool>(bIsSpatialJoin);
+
+  if ((aIs && bIs) || (!aIs && !bIs)) {
+    return std::nullopt;
+  }
+
+  SubtreePlan spatialSubtreePlan = aIsSpatialJoin ? a : b;
+  SubtreePlan otherSubtreePlan = aIsSpatialJoin ? b : a;
+
+  std::shared_ptr<Operation> op = spatialSubtreePlan._qet->getRootOperation();
+  SpatialJoin* spatialJoin = static_cast<SpatialJoin*>(op.get());
+
+  if (spatialJoin->isConstructed()) {
+    return std::nullopt;
+  }
```

Comment:
Ah, 
I now understand this...
You don't need `isConstructed` here,
but you need "is the single variabel that I want to bind here already bound"..

Considder for example

```
SELECT * { 
?x <p> ?y .  #1
?x2 <p2> ?y. #2
?y <maxDist:5> ?z. 
# irgendwas mit Z.
 }
```
The query planner has to pick one of the triples `1` or `2` as the left side of the spatial join. I think currently is allowed by this code to bind both of them. And this wouldn't be correct (one of the triples would be ignored/overwritten).

Reply:
i think it's not necessary to check, if the singe variable i want to bind is already bound because of the following reason: as soon as this one variable, here y, has been added, the variableToColumnMap returns one column, which only contains z. Therefore, the QueryPlanner can only join it with z variables. The behavior* of the VariableToColumnMap prevents this case.

*Behavior of VariableToColumnMap: If no child has been added, it returns the two variables of its children in it's map. As soon as one child has been added, it returns only one column, containing the missing child. When both children are finally added, it returns the "true" VariableToColumnMap, which contains the merged results from it's children